src/Config.php

```diff
@@ -71,50 +74,62 @@
  * For classes known from the definitions, a type preference might be the
  * better approach
  *
- * @see Laminas\Di\Resolver\ValueInjection A container to force injection of a value
- * @see Laminas\Di\Resolver\TypeInjection  A container to force looking up a specific type instance for injection
+ * @see \Laminas\Di\Resolver\ValueInjection A container to force injection of a value
+ * @see \Laminas\Di\Resolver\TypeInjection  A container to force looking up a specific type instance for injection
+ *
+ * @psalm-type TypeConfigArray = array{
+ *  typeOf?: class-string|null,
+ *  preferences?: array<string, string>|null,
+ *  parameters?: array<string, mixed>|null
```

Comment:
Do we actually allow `null` or is that something like a real-world example?

I would say, if `parameters` is provided (thanks to ?, it is not a required array key), only a map of strings should be provided. 🤷🏼‍♂️

Reply:
See `typeOf` remarks